app/controllers/publications/overview/shortlist.js

```diff
@@ -0,0 +1,127 @@
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import CONSTANTS from 'frontend-kaleidos/config/constants';
+
+export default class PublicationsOverviewShortlistController extends Controller {
+  @service publicationService;
+  @service router;
+  @service store;
+
+  @tracked isOpenNewPublicationModal;
+  pieceForPublication;
+  agendaitemForPublication;
+  decisionActivityForPublication;
+  caseForPublication;
+  mandateesForPublication;
+
+  getAgendaitem = async (piece) => {
+    const agendaitems = await piece.agendaitems;
+    let agendaitem;
+    for (let maybeAgendaitem of agendaitems) {
+      const agenda = await maybeAgendaitem.agenda;
+      const nextVersion = await agenda.nextVersion;
+      if (!nextVersion) {
+        agendaitem = maybeAgendaitem;
+        break;
+      }
+    }
+    return agendaitem;
+  }
```

Comment:
Prefer a query for this one.
We are loading a lot of data that we eventually don't need.

```
this.agendaitem = yield this.store.queryOne('agendaitem', {
      'filter[pieces][:id:]': this.args.piece.id,
      'filter[:has-no:next-version]': 't',
      sort: '-created',
```
This will yield the latest agendaitem linked to that piece.

Reply:
Because all the data is included, the only calls we actually do is getting `agendaitem.nextVersion` in the case that the nextVersion is null (and we could probably even work around this and have 0 calls instead).

Using a query means that we're doing this call over and over again, because `store.query(One)` isn't cached.
I'm okay with re-writing this but this will turn this route into one of many routes where we're making 10s or 100s of calls just to show a list.